### DISCUSSION TITLE
CLI audit command

### DISCUSSION THREAD

**Discussion by jeffsee55** [tinacms/tinacms#2279] [Ideas]
A command from `@tinacms/cli` to audit every content file and ensure it's adhering to the shape of the given schema. This work would be a precursor to a larger batch of work that would help users migrate content when schemas change, and would also likely help users coming from Forestry or Netlify CMS.

- Check that the file format matches the collection's `format` property, or if it's is not defined, make sure the files end with the default (`.md`).
- Check that the file can submit to GraphQL successfully

This should be done in a way that it can be run from the CLI as well as during CI from something like Github Actions.

Any other suggestions?

**Comment by jhuggett** [comment]
This would be great. 

Another issue I just ran into that could be solved in this is verifying that schemas are have correct properties. For example I just ran into an issue where my fields weren't matching what I wrote, until I realized that I had typed `optons` instead of `options`. Having this kinda thing checking whenever the dev server refreshes and spits out errors/warning would be helpful.

Half baked idea: I know we want to build at some point some kind of ui interface for modelling data. But what about starting with a TUI? Interactively build out you schema from the command line and it generates the schema code for you. For me it'd feel natural to use a tool like that as I'm developing the a site. Plus we can experiment and see what works best and then take any finding from there we/if we pursue a "Tina Studio" project
